packages/core/src/domain/telemetry/telemetry.ts

```diff
@@ -62,10 +74,16 @@ export function startTelemetry(telemetryService: TelemetryService, configuration
     !includes(TELEMETRY_EXCLUDED_SITES, configuration.site) && performDraw(configuration.telemetrySampleRate)
   telemetryConfiguration.telemetryConfigurationEnabled =
     telemetryConfiguration.telemetryEnabled && performDraw(configuration.telemetryConfigurationSampleRate)
+  telemetryConfiguration.telemetryUsageEnabled =
+    telemetryConfiguration.telemetryEnabled && performDraw(configuration.telemetryUsageSampleRate)
 
   const runtimeEnvInfo = getRuntimeEnvInfo()
   onRawTelemetryEventCollected = (rawEvent: RawTelemetryEvent) => {
-    if (telemetryConfiguration.telemetryEnabled) {
+    if (
+      telemetryConfiguration.telemetryEnabled &&
+      (rawEvent.type !== TelemetryType.configuration || telemetryConfiguration.telemetryConfigurationEnabled) &&
+      (rawEvent.type !== TelemetryType.usage || telemetryConfiguration.telemetryUsageEnabled)
+    ) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏This could probably be simplified with something like:
 ```ts

  const telemetryEnabled =
    !includes(TELEMETRY_EXCLUDED_SITES, configuration.site) && performDraw(configuration.telemetrySampleRate)

  const telemetryEnabledPerType = {
    [TelemetryType.log]: telemetryEnabled,
    [TelemetryType.configuration]: telemetryEnabled && performDraw(configuration.telemetryConfigurationSampleRate),
    [TelemetryType.usage]: telemetryEnabled && performDraw(configuration.telemetryConfigurationSampleRate),
  }

  const runtimeEnvInfo = getRuntimeEnvInfo()
  onRawTelemetryEventCollected = (rawEvent: RawTelemetryEvent) => {
    if (telemetryEnabledPerType[rawEvent.type!]) {
      ...
    }
  }
```